apollo-runtime-kotlin/src/commonMain/kotlin/com/apollographql/apollo/interceptor/BearerTokenInterceptor.kt

```diff
@@ -0,0 +1,61 @@
+package com.apollographql.apollo.interceptor
+
+import com.apollographql.apollo.ApolloException
+import com.apollographql.apollo.ApolloHttpException
+import com.apollographql.apollo.BearerTokenException
+import com.apollographql.apollo.api.ApolloExperimental
+import com.apollographql.apollo.api.Response
+import com.apollographql.apollo.network.HttpExecutionContext
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.flatMapConcat
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.retry
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+
+@ApolloExperimental
+class BearerTokenInterceptor(private val tokenProvider: TokenProvider) : ApolloRequestInterceptor {
+  val mutex = Mutex()
+
+  private fun <T> ApolloRequest<T>.withHeader(name: String, value: String): ApolloRequest<T> {
+    val httpRequestContext = executionContext[HttpExecutionContext.Request]
+        ?: HttpExecutionContext.Request(emptyMap())
+            .let {
```

Comment:
I feel like this `let` only add to the default branch with `emptyMap`. It should add to the existing context, shouldn't it?

Reply:
Oh my. Good catch, thanks !